dlt/common/storages/file_storage.py

```diff
@@ -251,3 +254,14 @@ def rmtree_del_ro(action: AnyFun, name: str, exc: Any) -> Any:
                 os.rmdir(name)
             else:
                 os.remove(name)
+
+    @staticmethod
+    def open_zipsafe(path: str, mode: str = "r", encoding: Optional[str] = None, **kwargs: Any) -> IO[Any]:
```

Comment:
this opens only for read. two options here:
- make it work with write (and use in data writers :))
- make sure that mode contains "r" with assert and maybe rename the method